client/state/document-head/selectors.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { getSelectedSiteId, isSiteSection } from 'state/ui/selectors';
+import { getSiteTitle } from 'state/sites/selectors';
+
+/**
+ * Returns the document title as set by the DocumentHead component or setTitle
+ * action.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Document title
+ */
+export function getTitle( state ) {
+	return state.documentHead.title;
+}
+
+/**
+ * Returns a count reflecting unread items.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Unread count (string because it can be e.g. '40+')
+ */
+export function getUnreadCount( state ) {
+	return state.documentHead.unreadCount;
+}
+
+/**
+ * Returns the formatted document title, based on the currently set title,
+ * unreadCount, and selected site.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Formatted title
+ */
+export function getFormattedTitle( state ) {
+	let title = '';
+
+	const unreadCount = getUnreadCount( state );
+	if ( unreadCount ) {
```

Comment:
Turns out `UpdateNotice` also uses the `40+` style `unreadCount`, so I opted to introduce another selector based on `getUnreadCount` — `getCappedUnreadCount`, which accepts a `cap` argument. Not sure it’s the most elegant solution, but at this point I fail to see any much more elegant solutions at hand.

FWIW, `connect`ing `UpgradeNotice` fortunately doesn’t seem to pose a problem, since it’s only used by `reader/stream`, which in turn is only used by the `reader/*-stream` family of subsections, which are all wrapped in a `Provider`.